packages/core/src/media/AudioManagerPool.ts

```diff
@@ -0,0 +1,89 @@
+import {Logger} from '../app';
+import {Sound} from '../scenes';
+import {AudioManager} from './AudioManager';
+
+export class AudioManagerPool {
+  private pools: {
+    am: AudioManager;
+    sounds: Sound[];
+    gain: GainNode;
+  }[] = [];
+
+  public constructor(private readonly logger: Logger) {}
+
+  public setupPools(sounds: Sound[]) {
+    this.pools = [];
+
+    for (const sound of sounds) {
+      const pool = this.pools.find(
+        pool =>
+          !pool.am.isInRange(sound.offset) &&
+          pool.am.getSource() === sound.audio,
+      );
+      if (pool !== undefined) {
+        pool.am.setOffset(sound.offset);
+        pool.am.setTrim(sound.start, sound.end);
+        pool.sounds.unshift(sound);
+      } else {
+        const am = new AudioManager(this.logger);
+        am.setOffset(sound.offset);
+        am.setTrim(sound.start, sound.end);
+        am.setSource(sound.audio);
+
+        let gain = undefined as unknown as GainNode;
+        am.setFilter((ctx, src) => {
+          gain = ctx.createGain();
+          src.connect(gain);
+          return gain;
+        });
+
+        this.pools.push({am, sounds: [sound], gain});
+      }
+    }
+  }
```

Comment:
I associate each sound with an audio manager beforehand because an AudioElement lagging would cause the audio manager not to be put back into the pool in time, which could make `pool.pop()` return undefined. We could create new managers on the fly, but I dislike this as
1. We are able to calculate how many managers are needed beforehand
2. If there are suddenly way too many managers in the pool due to user error, the user would have to reload the page to bring this back down or we would need some manager freeing logic too

I can remove the requirement the group sounds by their source however. I just assumed changing the source was a heavy operation. Then the sounds would just be grouped into non-overlapping sets

Reply:
Actually I did some testing and `setupPools` is kinda broken right now, AudioElement doesn't have the duration ready immediately so more often that not it just assumes all sounds have a duration of 0 which leads to cut off ends of sounds. I didn't notice before as I grouped them by source, and I didn't test with overlapping sounds of the same source.

Reply:
Actually, I could easily await for the metadata to be loaded in `setupPools`

Reply:
> we already need to add all sounds one by one to an AudioElement anyway to fully determine the duration of the sound

I think this should be a handled separately though, because:
1. We only need to do that once per source, the duration can then be cached for other sound instances
2. Once retrieved, the duration remains valid until a hot reload event. Scene recalculations themselves will not change the duration of an audio so we don't need to redo all this work each time `setupPools` is called.

Reply:
Imo the only question is whether changing the src of an AudioElement (after the browser has already fetched it) and seeking to a new position is slower than seeking alone. If it is indeed slower then segregating sounds so that the src of a manager never changes is a good idea. But otherwise it seems like more computation than it saves.

Reply:
I'll look into handling it seperately, for now I've fixed the existing pre-computed version (which now that i think about it i'm not that happy with)